packages/preview/src/browser/preview-link-normalizer.ts

```diff
@@ -0,0 +1,36 @@
+/********************************************************************************
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { Endpoint } from '@theia/core/lib/browser';
+
+@injectable()
+export class PreviewLinkNormalizer {
+
+    normalizeLink(documentUri: URI, link: string): string {
+        try {
+            const uri = new URI(link);
+            if (!uri.scheme) {
+                const location = documentUri.parent.resolve(link).path.toString();
+                return new Endpoint({ path: 'mini-browser/' + location }).getRestUrl().toString();
```

Comment:
@kittaakos, I reverted the async change to FileLocationMapper, but also haven't found a nice solution to make use of it. WDYT about a specialized `MiniBrowserEndpoint`? Constructor could take a location. It would be a pragmatic solution to get rid of repeating the path pattern here.

Reply:
Please, have a look.